server/modules/automations.js

```diff
@@ -59,3 +77,35 @@ export const createOrUpdateEmailAutomation = automationDetails => EmailAutomatio
  */
 // eslint-disable-next-line max-len
 export const createOrUpdateFreckleAutomation = automationDetails => FreckleAutomation.upsertById(automationDetails);
+
+/**
+ * @func creatOrUpdatePartnerRecord
+ * @desc create or update a partner record in the database
+ *
+ * @param {object} partnerDetails The partner details
+ * @param {string} partnerDetails.name Name of partner
+ * @param {string} partnerDetails.partnerId Id of partner from allocations
+ * @param {string} partnerDetails.freckleProjectId The partner freckle project ID
+ * @param {object} partnerDetails.slackChannels The partner slack channels
+ * @param {string} partnerDetails.slackChannels.internal The partner internal slack channel
+ * @param {string} partnerDetails.slackChannels.general The partner general slack channel
+ *
+ * @returns {Promise} Promise that resolves to the created/updated partner record
+ */
+export const creatOrUpdatePartnerRecord = async (partnerDetails) => {
+  const { partnerId } = partnerDetails;
+  const existingRecord = await db.Partner.find({ where: { partnerId } });
+  if (existingRecord) {
+    return db.Partner.update(partnerDetails, { where: { partnerId } });
+  }
+  return db.Partner.create(partnerDetails);
+};
+
+/**
+ * @func getPartnerRecord
+ * @desc Get a partner record from the database
+ *
+ * @param {string} partnerId The id of the partner(from the placement data).
+ * @returns {Promise} Promise that resolves to the found partner record.
+ */
+export const getPartnerRecord = partnerId => Partner.find({ where: { partnerId } });
```

Comment:
I think this function has been deprecated too. Kindly rebase from develop

Reply:
rebased successfully.